distributed/core.py

```diff
@@ -1544,6 +1544,8 @@ async def _connect(self, addr: str, timeout: float | None = None) -> Comm:
                 self._connecting_count -= 1
         except asyncio.CancelledError:
             current_task = asyncio.current_task()
+            if sys.version_info >= (3, 11):
+                current_task.uncancel()
```

Comment:
This isn't quite the right usage for uncancel. It should only be called if `self._reasons.pop(current_task` is set and then CancelledError should be raised if uncancel() != 0

Reply:
One option is to use

```python
try:
    async asyncio.timeout(math.inf) as scope:
        await ...
except TimeoutError:
    raise CommClosedError(...)
```

And then `scope.reschedule(-1)` to cancel the task

Reply:
yeah, but I won't rewrite the entire code for a py3.11 only option

Reply:
I already started on it here: https://github.com/dask/distributed/pull/8109 :stuck_out_tongue_closed_eyes:

Reply:
I think it's worth it as we can skip a load of bookkeeping for external/internal cancellation on 3.11, and user code isn't really supposed to call `.uncancel()`